File: java/sivitas.java
/*
Saya [Elsa Nabiilah] [2108805] mengerjakan LATIHAN 2 dalam mata kuliah Desain Pemrograman Berorientasi Objek
untuk keberkahanNya maka saya tidak melakukan kecurangan seperti yang telah dispesifikasikan.
Aamiin.
*/

//membuat public class dengan ibu mahasiswa
public class sivitas extends Mahasiswa {
    // deklarasi variabel private
    private String univ;
    private String email;

    // initialize
    public sivitas() {
        this.univ = "";
        this.email = "";

    }

    public sivitas(String univ, String email) {
        this.univ = univ;
        this.email = email;
    }

    // getter setter method
    public String getuniv() {
        return this.univ;
    }

    public void setuniv(String univ) {
        this.univ = univ;
    }

    public String getemail() {
        return this.email;
    }

    public void setemail(String email) {
        this.email = email;
    }

    // membuat fungsi sesuai atribut
    void asal_univ() {
        System.out.println("Asal Univ     : " + this.univ);
    }

    void email() {
        System.out.print("email         : " + this.email);
    }

}

File: cpp/sivitas.cpp
#include <bits/stdc++.h>
/*
Saya [Elsa Nabiilah] [2108805]
mengerjakan LATIHAN 2 dalam mata kuliah Desain Pemrograman Berorientasi Objek
untuk keberkahanNya maka saya tidak melakukan kecurangan seperti yang telah dispesifikasikan.
Aamiin.
*/
using namespace std;

#include "Mahasiswa.cpp"

class sivitas : public Mahasiswa
{
    // sivitas menjadi child dari mahasiswa karena sesuai dengan
    // hakikatnya setiap mahasiswa pasti memiliki atribut/identitas kampus seperti
    // asal universitas dan email

    //  deklarasi variabel private
private:
    string univ;
    string email;

    // conctructor
public:
    sivitas()
    {
    }

    sivitas(string univ, string email)
    {
        this->univ = univ;
        this->email = email;
    }

    // getter setter method
    string getuniv()
    {
        return this->univ;
    }

    void setuniv(string univ)
    {
        this->univ = univ;
    }

    string getemail()
    {
        return this->email;
    }

    void setemail(string email)
    {
        this->email = email;
    }

    // membuat fungsi menampilkan per atribut
    void printuniv()
    {
        cout << "Asal Univ     : " << this->getuniv() << endl;
    }
    void printmail()
    {
        cout << "Email Univ    : " << this->getemail() << endl;
    }

    ~sivitas() {}
};
